colin/core/checks/fmf_check.py

```diff
@@ -0,0 +1,148 @@
+            ref_item_name = node.name.rsplit("@", 1)[1]
+            # match item what does not contain @ before name, otherwise it
+            # match same item
+            reference_node = patterntree.search("[^@]%s" % ref_item_name)
+            logger.debug("MERGING: %s @ %s", node.name, reference_node.name)
+            if not reference_node:
+                raise ValueError("Unable to find reference for node: %s  via name search: %s" %
+                                 (node.name, ref_item_name))
+            node.merge(parent=reference_node)
+
+        self.__remove_append_items(whole=whole)
+
+    def search(self, name):
+        """ Search node with given name based on reqexp"""
+        for node in self.climb():
+            if re.search(name, node.name):
+                return node
+        return None
+
+
+class FMFCaseLoader(object):
+    """
+    search and load FMF metadata
+    """
+
+    def __init__(self, name_id, path=None):
+        """
+        Case loader metatada init, it try to gather metadata form FMF based on name identifier
+        :param name_id:
+        :param path:
+        """
+        self.metadata = self.__receive_fmf_metadata(
+            path or get_checks_path(), name=name_id)
+        self.name = self.get_name()
+
+    def __receive_fmf_metadata(self, fmfpath, name=None, object_list=False):
+        """
+        internal method, search name in metadata in fmfpath
+
+        :param fmfpath: path to filesystem
+        :param name: str - name as pattern to search (substring)
+        :param object_list: bool, if true, return whole list of found items
+        :return: Tree Object or list
+        """
+        output = {}
+        fmf_tree = ExtendedTree(fmfpath)
+        logger.debug("get FMF metadata for test (path:%s name=%s)", fmfpath, name)
+        items = [x for x in fmf_tree.climb(
+        ) if name in x.name and "@" not in x.name]
+        if object_list:
+            return items
+        if len(items) == 1:
+            output = items[0]
+        elif len(items) > 1:
+            raise Exception("There is more FMF test metadata for item by name:{}({}) {}".format(
+                name, len(items), [x.name for x in items]))
+        elif not items:
+            raise Exception("Unable to get FMF metadata for: {}".format(name))
+        return output
+
+    def get_name(self, full=False):
+        """
+        return FMF Name of item
+        :param full: if True return full path identifier
+        :return: str - name identifier
+        """
+        out = self.metadata.name
+        if not full:
+            out = self.metadata.name.rsplit("/", 1)[-1]
+        return out
+
+
+class FMFAbstractCheck(AbstractCheck):
+    """
+    Abstract class for checks and loading metadata from FMF format
+    """
+    metadata = None
+    name = None
+
+    @classmethod
+    def get_metadata(cls, name, path=None):
+        """
+        Very important method what returns tuple for object initialization
```

Comment:
Get rid of documentation like 'Very important method'. It is enough to say. 'Method returns ..'